services/api-server/src/simcore_service_api_server/api/routes/jobs.py

```diff
@@ -0,0 +1,178 @@
+import logging
+from contextlib import contextmanager
+from typing import Callable, List
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, HTTPException
+from starlette import status
+
+from ...models.schemas.solvers import Job, JobInput, JobOutput, JobStatus, KeyIdentifier
+from ..dependencies.application import get_reverse_url_mapper
+from .jobs_faker import the_fake_impl
+from .solvers import router as solvers_router
+
+logger = logging.getLogger(__name__)
+
+
+router = APIRouter()
+
+
+@contextmanager
+def errors_mapper(to_not_found=KeyError):
+    try:
+        yield
+    except to_not_found as err:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND) from err
+
+
+## JOBS ---------------
+#
+# - Similar to docker container's API design (container = job and image = solver)
+#
+
+
+@solvers_router.get("/{solver_id}/jobs/")
+async def list_jobs(
+    solver_id: UUID,
+    url_for: Callable = Depends(get_reverse_url_mapper),
+):
+    """ List of all jobs with a given solver """
+    # TODO: add pagination
+    # similar to ps = process status
+    return [
+        Job(
+            solver_url=url_for(
+                "get_solver_by_id",
+                solver_id=job["solver_id"],
+            ),
+            inspect_url=url_for("inspect_job", job_id=job["job_id"]),
+            outputs_url=url_for("list_job_outputs", job_id=job["job_id"]),
+            **job,
+        )
+        for job in the_fake_impl.iter_jobs(solver_id)
+    ]
+
+
+# pylint: disable=dangerous-default-value
+@solvers_router.post("/{solver_id}/jobs/", response_model=Job)
+async def create_job(
+    solver_id: UUID,
+    inputs: List[JobInput] = [],
+    url_for: Callable = Depends(get_reverse_url_mapper),
+):
+    """Creates a job for a solver with given inputs.
+
+    NOTE: This operation does **not** start the job
```

Comment:
As mentioned yesterday. I am not sure if this is useful. But users will have the last word